nautobot/extras/models/groups.py

```diff
@@ -17,16 +18,153 @@
 from nautobot.core.forms.fields import DynamicModelChoiceField
 from nautobot.core.forms.widgets import StaticSelect2
 from nautobot.core.models import BaseManager, BaseModel
-from nautobot.core.models.generics import OrganizationalModel
+from nautobot.core.models.generics import OrganizationalModel, PrimaryModel
 from nautobot.core.utils.config import get_settings_or_config
 from nautobot.core.utils.lookup import get_filterset_for_model, get_form_for_model
 from nautobot.extras.choices import DynamicGroupOperatorChoices
 from nautobot.extras.querysets import DynamicGroupMembershipQuerySet, DynamicGroupQuerySet
-from nautobot.extras.utils import extras_features
+from nautobot.extras.utils import extras_features, FeatureQuery
 
 logger = logging.getLogger(__name__)
 
 
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "webhooks",
+)
+class StaticGroup(PrimaryModel):
+    """A statically defined (as opposed to dynamically calculated) group of objects sharing a common content-type."""
+
+    name = models.CharField(max_length=CHARFIELD_MAX_LENGTH, unique=True)
+    description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
+    content_type = models.ForeignKey(
+        to=ContentType,
+        on_delete=models.CASCADE,
+        related_name="static_groups",
+        limit_choices_to=FeatureQuery("static_groups"),
+        help_text="The type of object contained in this Static Group.",
+    )
+
+    clone_fields = ["content_type"]
+    is_static_group_associable_model = False
```

Comment:
Would it be trivial to make static groups a static group associable model? I could see it being useful for something like this:

- `StaticGroup(content_type = StaticGroup, name="Change Window 1234")`
  - `StaticGroup(content_type = Device, name="Change Window 1234 Devices")`
  - `StaticGroup(content_type = Controller, name="Change Window 1234 Controllers")`

Then an app could be configured to only make changes to the Devices and Controllers in the child groups during the change window.

Reply:
Interesting idea! I was going by the fact that StaticGroups are not (per #5472) intended to be self-nestable at this time (Group A of Devices containing Group B also of Devices and therefore implicitly including all of Group B's members), but it's an interesting idea to use a static group to relate groups of otherwise unrelated objects. I'm not sure I'm convinced that this is needed at this time though, couldn't the app just point to the "child" groups individually?

Reply:
> couldn't the app just point to the "child" groups individually?

Definitely, I'm just trying to think of why we wouldn't want to support this use case. Are we thinking of making StaticGroups some kind of TreeModel or supporting multiple content types in the future?

Reply:
I think we're at least considering the TreeModel approach at some point yeah. I think I'd be concerned about user expectations of what a static-group-of-static-groups "means" as far as inheritance goes and such, especially given the example we've already got with dynamic groups containing other dynamic groups. I'd prefer to avoid that for now; as you pointed out it *should* be a simple change if we decide later that it's something we need.